src/ClientMessage.ts

```diff
@@ -1,206 +1,189 @@
-///<reference path="../typings/node/node.d.ts" />
-///<reference path="../typings/long/long.d.ts" />
 /* tslint:disable:no-bitwise */
-/*
- Client Message is the carrier framed data as defined below.
- Any request parameter, response or event data will be carried in the payload.
- 0                   1                   2                   3
- 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
- +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
- |R|                      Frame Length                           |
- +-------------+---------------+---------------------------------+
- |  Version    |B|E|  Flags    |               Type              |
- +-------------+---------------+---------------------------------+
- |                       CorrelationId                           |
- |                                                               |
- +---------------------------------------------------------------+
- |                        PartitionId                            |
- +-----------------------------+---------------------------------+
- |        Data Offset          |                                 |
- +-----------------------------+                                 |
- |                      Message Payload Data                    ...
- |                                                              ...
- */
```

Comment:
I can think of some benefits like;
1. This is source code documentation not spec. So we are showing what we implement. Spec allows to add custom data in a client message using the Data-offset field so the actual implementation may vary. The documentation should display what is actually implemented. 
2. Spec itself may change and add more fields to the client message. So the documentation will help the developers to see what is actually implemented.
3. Documented code is better than non-documented code and give more confidence to community I believe.

For the second comment that the doc is outdated; It is not. Correlation Id occupy two lines which has 32bit each.

Note: We borrowed the above notation from Martin Thompson's Aeron project. 
https://github.com/real-logic/Aeron/blob/master/aeron-client/src/main/java/uk/co/real_logic/aeron/protocol/HeaderFlyweight.java